### ISSUE TITLE
Question: Training student on unlabeled images

### ISSUE THREAD

**Issue by schirrmacher** [DepthAnything/Depth-Anything-V2#24]
In the paper you mention the following:

> We first train the most capable teacher on precise synthetic images. Then, to mitigate the distribution shift and limited diversity of synthetic data, we annotate unlabeled real images with the teacher. Finally, we train student models on high-quality pseudo-labeled images.

I fully understand the idea. But how can the teacher work good enough on unlabeled images when it is based on synthetic data? How do you guarantee the quality of the pseudo-labeled images? I do not see how or where you bridge the synthetic gap...

**Comment by LiheYoung** [comment]
> Q1: How can the teacher work well enough on unlabeled images when it is based on synthetic data?

We compare different pre-trained encoders in Section 3, and find the DINOv2-Giant-based depth model transfers well enough from synthetic training images to real test images. This is supported by both qualitative (Figure 5 and 17) and quantitative (Table 5 last row) results.

> Q2: How to guarantee the quality of the pseudo-labeled images?

At least two factors guarantee the pseudo label quality. First, as aforementioned, we empirically find the DINOv2-Giant-based depth model can serve as a reliable teacher model. Second, we discard 10% largest-loss regions when training student models on pseudo-labeled real images.

> Q3: How is the domain gap between synthetic training images and real test images bridged?

The student models are purely trained on high-quality pseudo-labeled real images. Briefly, it will learn the diverse real-image distributions with the pseudo label. For more in-depth analysis, you may refer to the research field of unsupervised domain adaptation.

**Comment by luochunfen** [comment]
> > Q1: How can the teacher work well enough on unlabeled images when it is based on synthetic data?
> 
> We compare different pre-trained encoders in Section 3, and find the DINOv2-Giant-based depth model transfers well enough from synthetic training images to real test images. This is supported by both qualitative (Figure 5 and 17) and quantitative (Table 5 last row) results.
> 
> > Q2: How to guarantee the quality of the pseudo-labeled images?
> 
> At least two factors guarantee the pseudo label quality. First, as aforementioned, we empirically find the DINOv2-Giant-based depth model can serve as a reliable teacher model. Second, we discard 10% largest-loss regions when training student models on pseudo-labeled real images.
> 
> > Q3: How is the domain gap between synthetic training images and real test images bridged?
> 
> The student models are purely trained on high-quality pseudo-labeled real images. Briefly, it will learn the diverse real-image distributions with the pseudo label. For more in-depth analysis, you may refer to the research field of unsupervised domain adaptation.

Could you give more details about discarding 10% largest-loss regions when training student models on pseudo-labeled real images ?  I can't figure out how to implement this trick.
